packages/element/src/components/interpolate/interpolate.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Internal dependencies
+ */
+import {
+	createInterpolateString,
+	createInterpolateElement,
+	getInterpolateMap,
+} from './utils';
+
+/**
+ * A component that allows for easier interpolation of values.
+ *
+ * This is useful for dynamic type string-like elements with embedded components
+ * and values.
+ *
+ * @example
+ *
+ * ```js
+ * const MyComponent = ( { url } ) => {
+ *   return <Interpolate foo={ 'bar' } >
+ *       This is a string with a <a href={ url }>linked <em>item</em></a> and a custom value: %%foo%%
+ *   </Interpolate>;
+ * }
+ * ```
+ *
+ * @param {Object} props
+ * @return {WPElement} An element.
+ * @constructor
+ */
+const Interpolate = ( props ) => {
+	return createInterpolateElement(
+		createInterpolateString( props.children, props ),
```

Comment:
As noted in the pull description, right now this isn't localizing anything, but it does demonstrate how a localizable string can be built using this new helper method.  I think if we decide to expose the nicer api via a component, we should use babel to transform the `props.children` to the interpolation string and output it in the built code.  I haven't written a babel plugin yet so that's something I might need help with.